### PR TITLE
Get role data directly from CouchDB

### PR DESCRIPTION
**Pull request by skazi0**
**Why is this change necessary?**
The performance of `crowbarctl node list` and dashboard needs improvement.

**How does it address the issue?**
This improves performance of node/role operations on large installations.
Chef API is skipped resulting in less JSON (de)serialization.

**Is there additional information worth sharing like links to a Trello
card, bug references, testing advice or dependencies to other pull
requests?**
https://trello.com/c/kQ1eLgYY/122-evaluate-performance-of-admin-node-on-scale-cloud

### CODE REVIEW

**Comment by skazi0** [comment]
[Evaluate performance of admin node on scale cloud](https://trello.com/c/kQ1eLgYY/122-evaluate-performance-of-admin-node-on-scale-cloud)

File: crowbar_framework/app/models/role_object.rb [hunk 1]
```diff
@@ -206,7 +206,8 @@ def self.find_roles_by_search(search)
 
   def self.find_role_by_name(name)
     begin
-      return RoleObject.new Chef::Role.load(name)
+      # get role directly from CouchDB, skip Chef API
+      return RoleObject.new Chef::Role.cdb_load(name)
     rescue Errno::ECONNREFUSED => e
       raise Crowbar::Error::ChefOffline.new
     rescue Net::HTTPServerException => e
```
Comment (vuntz) [inline crowbar_framework/app/models/role_object.rb]: Actually: don't we need to check for some other exception (isn't there one if the role with that name doesn't exist)?

File: crowbar_framework/app/models/role_object.rb [hunk 2]
```diff
@@ -206,7 +206,8 @@ def self.find_roles_by_search(search)
 
   def self.find_role_by_name(name)
     begin
-      return RoleObject.new Chef::Role.load(name)
+      # get role directly from CouchDB, skip Chef API
+      return RoleObject.new Chef::Role.cdb_load(name)
```
Comment (Itxaka) [inline crowbar_framework/app/models/role_object.rb]: as we are hitting the database, we wont get those errors anymore no?

File: crowbar_framework/app/models/role_object.rb (hunk 1 above)
Comment (skazi0) [reply crowbar_framework/app/models/role_object.rb]: From the code it looks like we should get the same exceptions as both calls use `Chef::REST` behind the scenes just hitting different backends.
I'm wondering if it's worth introducing `Crowbar::Error::CouchDBOffline` exception for these new cases...

File: crowbar_framework/app/models/role_object.rb (hunk 2 above)
Comment (skazi0) [reply crowbar_framework/app/models/role_object.rb]: It's REST interface so I guess we should get the same kind of errors.

Comment (Itxaka) [reply crowbar_framework/app/models/role_object.rb]: +1

File: crowbar_framework/app/models/role_object.rb (hunk 1 above)
Comment (vuntz) [reply crowbar_framework/app/models/role_object.rb]: @skazi0 I'd feel slightly more comfortable if we know for sure that a non-existing role raises the same 404 Net::HTTPServerException :-)

Comment (skazi0) [reply crowbar_framework/app/models/role_object.rb]: @vuntz as always... you're right. CouchDB returns empty result-set if nothing is found. Chef API was transforming it to 404.

**Pull request merged**